consensus/aggregator.go

```diff
@@ -0,0 +1,43 @@
+package consensus
+
+import (
+	"fmt"
+
+	"github.com/gammazero/workerpool"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/voteaggregator"
+	"github.com/onflow/flow-go/consensus/hotstuff/votecollector"
+	"github.com/onflow/flow-go/consensus/recovery"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// NewVoteAggregator creates new VoteAggregator and recover the Forks' state with all pending block
+func NewVoteAggregator(
+	log zerolog.Logger,
+	finalized *flow.Header,
+	pending []*flow.Header,
+	notifier hotstuff.Consumer,
+	forks hotstuff.Forks,
+	validator hotstuff.Validator,
+	voteProcessorFactory hotstuff.VoteProcessorFactory,
+) (hotstuff.VoteAggregator, error) {
+
+	createCollectorFactoryMethod := votecollector.NewStateMachineFactory(log, notifier, voteProcessorFactory.Create)
+	voteCollectors := voteaggregator.NewVoteCollectors(finalized.View, workerpool.New(4), createCollectorFactoryMethod)
+
+	// initialize the vote aggregator
+	aggregator, err := voteaggregator.NewVoteAggregator(log, notifier, finalized.View, voteCollectors)
+	if err != nil {
+		return nil, fmt.Errorf("could not create vote aggregator: %w", err)
+	}
+
+	// recover the hotstuff state, mainly to recover all pending blocks in Forks
+	err = recovery.Participant(log, forks, aggregator, validator, finalized, pending)
+	if err != nil {
+		return nil, fmt.Errorf("could not recover hotstuff state: %w", err)
+	}
```

Comment:
To me, it would be more intuitive to leave the recovery in `NewParticipant`, specifically here:  https://github.com/onflow/flow-go/blob/b4f763879799deb3993e744dbba04941a217934d/consensus/participant.go#L64-L65
Otherwise, we remove the nice symmetry with the Follower: https://github.com/onflow/flow-go/blob/b4f763879799deb3993e744dbba04941a217934d/consensus/follower.go#L31-L35 and the code becomes more erratic.  

### Suggestion: 
1. We remove the parameters `pendingBlocks` from the `VoteAggregator`'s constructor. Technically, we could just initialize it with view zero and prune later. 
2. Thereby, we avoid the need to have the pending blocks and can replace https://github.com/onflow/flow-go/blob/b4f763879799deb3993e744dbba04941a217934d/cmd/consensus/main.go#L588 by
    ```golang
    finalizedBlock, err := state.Final().Head()
    ```
3. our pruning code is super efficient for a empty `VoteCollectors`, so we can just prune here: https://github.com/onflow/flow-go/blob/b4f763879799deb3993e744dbba04941a217934d/consensus/participant.go#L64
4. We move the recovery code back into `NewParticipant` to line 65 https://github.com/onflow/flow-go/blob/b4f763879799deb3993e744dbba04941a217934d/consensus/participant.go#L64-L65

### Caution: our pruning convention is inconsistent and needs to be fixed:
* `VoteCollectors` removes elements _below_ the given view number: https://github.com/onflow/flow-go/blob/b4f763879799deb3993e744dbba04941a217934d/consensus/hotstuff/voteaggregator/vote_collectors.go#L102-L103
* while `VoteAggregator` claims to include the specified view: https://github.com/onflow/flow-go/blob/b4f763879799deb3993e744dbba04941a217934d/consensus/hotstuff/voteaggregator/vote_aggregator.go#L214-L221 (but the implementation doesn't really do so). 

Overall, I would suggest to stick with the existing convention: 
> Prune prunes all blocks UP TO but NOT INCLUDING the given value

Our `VoteAggregator` implementation already does this, but the method documentation is different.